src/Options.php

```diff
@@ -1342,4 +1166,37 @@ private function validateContextLinesOption(?int $contextLines): bool
     {
         return $contextLines === null || $contextLines >= 0;
     }
+
+    /**
+     * Calls the resolve method of the internal resolver with a logger so that
+     * validation failures can be logged and investigated.
+     *
+     * @param array<string, mixed> $options
+     *
+     * @return array<string, mixed>
+     */
+    private function resolveWithLogger(array $options = []): array
+    {
+        /**
+         * @var LoggerInterface $logger
+         */
+        $logger = $options['logger'] ?? $this->getLoggerOrNullLogger();
+
+        return $this->resolver->resolve($options, $logger);
+    }
+
+    /**
+     * Merges the passed options with the current options and resolves them.
+     * The result is stored back onto the class field.
+     *
+     * @param array<string, mixed> $override
+     */
+    private function updateOptions(array $override = []): self
```

Comment:
We might need to make this public (and `@internal`) if we end up having a use for it in the SDKs. But `private` is fine for now.